lib/common/options.js

```diff
@@ -132,6 +132,23 @@ Options.from = function (_options) {
     options.store = true
   }
 
+  // Set the name for global variables - should contain only letters
+  options.globalName = (_.isString(options.globalName) && /^[a-zA-Z]+$/.test(options.globalName))
+    ? options.globalName.toLowerCase()
+    : 'nuxt'
+
+  // Call globalName option function setters
+  for (const globalNameOption in options.globals) {
+    if (typeof options.globals[globalNameOption] === 'function') {
+      options.globals[globalNameOption] = options.globals[globalNameOption](options.globalName)
+    }
+  }
+  for (const globalCallbackOption in options.globals.callbacks) {
```

Comment:
What if flatten `globals.callbacks` into `globals.readyCallback`, etc. This way we can eliminate this for loop. (If more root keys like this are added we have to repeat this logic)

Reply:
@pi0 I was a little uneasy about it because I feel `callbacks.something` is cleaner than `somethingCallback`, given we've been following an hierarchical style, so it made sense to try and group these things semantically. But as long as it's only a couple of callbacks, I see no harm. Will change before we merge.